### PR TITLE
[WIP] Shipping info

### PR DESCRIPTION
**Pull request by benzguo**
@jflinter @bdorfman 

This is a WIP demo of shipping method collection, for initial design/API review.

### Intentionally omitted functionality
- There are only two `STPShippingType`s, `Shipping` and `Delivery` (`Delivery` doesn't ask for shipping method). I think we can omit PassKit's `StorePickup` and `ServicePickup` for now.
- `STPShippingAddressVC` and `STPShippingMethodsVC` are private, and can only be used via `STPPaymentContext`.

### Todos
- [ ] Set `shippingContact` on `PKPaymentRequest`
- [ ] Dynamically update shipping methods when changing shipping address in Apple Pay
- [ ] Throw an exception if `requiredShippingAddressFields` is `None` and you try to present the shipping methods VC
- [ ] analytics

### Proposed API changes / refactoring

I'm planning on doing these in subsequent PRs into the `9.0.0` branch.

- [ ] [Public] Rename `STPPaymentActivityIndicatorView` -> `STPActivityIndicatorView` (or make it private)
- [ ] [Public] Rename `STPBillingAddressFields` -> `STPAddressFields`
- [ ] [Public] Rename `STPAddress` `applePayAddressFieldsFromBillingAddressFields` -> `applePayAddressFieldsFromAddressFields`
- [ ] [Private] Rename `STPAddressViewModel` `initWithRequiredBillingFields` -> `initWithRequiredFields`
- [ ] [Private] Make a base view controller superclass to de-dupe some shared code

### CODE REVIEW

File: Example/Stripe iOS Example (Simple)/CheckoutViewController.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ class CheckoutViewController: UIViewController, STPPaymentContextDelegate {
 
     // 1) To get started with this demo, first head to https://dashboard.stripe.com/account/apikeys
     // and copy your "Test Publishable Key" (it looks like pk_test_abcdef) into the line below.
-    let stripePublishableKey = ""
+    let stripePublishableKey = "pk_test_DhW2G4j13BMRIC7gd55hZeW3"
```
Comment (bdorfman) [inline Example/Stripe iOS Example (Simple)/CheckoutViewController.swift]: You probably didn't  mean to commit this stuff?

File: Stripe/STPPaymentContextAmountModel.m [hunk 2]
```diff
@@ -35,28 +35,50 @@ - (instancetype)initWithPaymentSummaryItems:(NSArray<PKPaymentSummaryItem *> *)p
...
         PKPaymentSummaryItem *totalItem = [PKPaymentSummaryItem summaryItemWithLabel:companyName
-                                                                              amount:amount];
-        return @[totalItem];
-        
+                                                                              amount:total];
+        NSMutableArray *items = [@[totalItem] mutableCopy];
+        if (shippingItem != nil) {
+            [items insertObject:shippingItem atIndex:0];
+        }
+        return items;
```
Comment (bdorfman) [inline Stripe/STPPaymentContextAmountModel.m]: Should technically return `items.copy` so not mutable return value.

File: Stripe/STPShippingAddressViewController.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import "STPTheme.h"
+#import "STPPaymentContext.h"
+
+@protocol STPShippingAddressViewControllerDelegate;
+
+@interface STPShippingAddressViewController : UIViewController
```
Comment (bdorfman) [inline Stripe/STPShippingAddressViewController.h]: We should add nullability specifiers for all new classes going forward.

**Comment by bdorfman** [comment]
Looks fine to me.

**Comment by jflinter** [comment]
As a meta-note, I'd prefer a slightly different order-of-operations around building/releasing this (this is something I wish we had done for the original `STPPaymentContext` work, and I'd like to learn from it). Basically, I'd rather you put together an `STPShippingAddressViewController` class first, and then in a subsequent PR add `STPPaymentContext` support for it. I think it's important that those classes work on their own, and this will basically force us to do that.

**Comment by benzguo** [comment]
@jflinter I've split out a new PR without the `STPPaymentContext` changes here: #446 

I'm going to close this one – I'll open another PR with `STPPaymentContext` updates once #446 is in `9.0.0`.

**Pull request closed without merging**